app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -2,16 +2,7 @@
 class OmniauthCallbacksController < Devise::OmniauthCallbacksController
   def github
     user = User.find_or_create_for_github_oauth(request.env['omniauth.auth'])
-
-    if user.just_created? || !user.valid? || user.github_import
-      sign_in user
-      redirect_to edit_user_path(user, welcome: true)
-    elsif user.persisted?
-      flash[:notice] = I18n.t 'devise.omniauth_callbacks.success', kind: 'Github'
-      sign_in_and_redirect user, event: :authentication
-    else
-      session['devise.github_data'] = request.env['omniauth.auth']
-      redirect_to new_user_registration_url
-    end
```

Comment:
All our users are either persisted on :create, or not confirmed.  So I can't think of a situation where the third clause is triggered? All the tests pass without this clause, and I can't think of a reason we need it.